server/src/test/java/org/opensearch/action/admin/cluster/node/stats/NodeStatsTests.java

```diff
@@ -1065,4 +1088,423 @@ private static RemoteTranslogTransferTracker.Stats getRandomRemoteTranslogTransf
+
+                Map<String, Object> xContentMap = xContentBuilderToMap(builder);
+                LinkedHashMap indicesStatsMap = (LinkedHashMap) xContentMap.get(NodeIndicesStats.StatsLevel.INDICES.getRestName());
+
+                switch (level) {
+                    case NODE:
+                        assertFalse(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.INDICES.getRestName()));
+                        assertFalse(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.SHARDS.getRestName()));
+                        break;
+                    case INDICES:
+                        assertTrue(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.INDICES.getRestName()));
+                        assertFalse(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.SHARDS.getRestName()));
+                        break;
+                    case SHARDS:
+                        assertFalse(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.INDICES.getRestName()));
+                        assertTrue(indicesStatsMap.containsKey(NodeIndicesStats.StatsLevel.SHARDS.getRestName()));
+                        break;
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+        });
+    }
+
+    public void testNodeIndicesStatsWithAndWithoutAggregations() throws IOException {
+
+        CommonStatsFlags commonStatsFlags = new CommonStatsFlags(
+            CommonStatsFlags.Flag.Docs,
+            CommonStatsFlags.Flag.Store,
+            CommonStatsFlags.Flag.Indexing,
+            CommonStatsFlags.Flag.Completion,
+            CommonStatsFlags.Flag.Flush,
+            CommonStatsFlags.Flag.FieldData,
+            CommonStatsFlags.Flag.QueryCache,
+            CommonStatsFlags.Flag.Segments
+        );
+
+        int numberOfIndexes = randomIntBetween(1, 3);
+        List<Index> indexList = new ArrayList<>();
+        for (int i = 0; i < numberOfIndexes; i++) {
+            Index index = new Index("test-index-" + i, "_na_");
+            indexList.add(index);
+        }
+
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        HashMap<Index, List<IndexShardStats>> statsByShards = createRandomShardByStats(indexList);
+        Map<Index, CommonStats> statsByIndex = createStatsByIndex(statsByShards);
+
+        final MockNodeIndicesStats nonAggregatedNodeIndicesStats = new MockNodeIndicesStats(
+            new CommonStats(commonStatsFlags),
+            statsByShards,
+            new SearchRequestStats(clusterSettings)
+        );
+
+        commonStatsFlags.setIncludeIndicesStatsByLevel(true);
+
+        Arrays.stream(NodeIndicesStats.StatsLevel.values()).forEach(level -> {
+            MockNodeIndicesStats aggregatedNodeIndicesStats;
+            if (level.equals(NodeIndicesStats.StatsLevel.INDICES) || level.equals(NodeIndicesStats.StatsLevel.NODE)) {
```

Comment:
Can we please use CONST.equals(var) for comparisons?

Reply:
Updated in the revision